backend/ipmi.pm

```diff
@@ -41,8 +41,13 @@ sub ipmitool ($self, $cmd, %args) {
     }
     chomp $stdout;
     chomp $stderr;
+    unless ($ret) {
+        # Don't show ipmi password in error output
+        my $secreted_cmd = join(' ', @cmd);
+        $secreted_cmd =~ s/-P\s+\w+/-P <secret>/;
+        die "$secreted_cmd: $stderr";
+    }
```

Comment:
As we already have some special handling for masking secrets I wonder, @cfconrad do you have an idea how something like https://github.com/os-autoinst/os-autoinst/blob/af9bd7805f9b3b51b3516c0334f72cb162605079/bmwqemu.pm#L209 can be used here without duplicating the implementation?

Reply:
This code parses the `-P` argument from `ipmi`. The secret hiding in `log_call` in `bmwqemu.pm` replaces values. I'm not sure about sharing code, but it should probably behave the same way?

Reply:
At the very least we should use the same text `[masked]` and remove the die-message :)

Reply:
Thanks. changed to '[masked]' to behave the same way. Now the new error output is :
```
[2022-07-13T21:21:44.454344-04:00] [info] ::: backend::baseclass::die_handler: Backend process died, backend errors are reported below in the following lines:
  ipmitool -I lanplus -H 10.67.135.53 -U tester -P [masked] mc guid: Error: Unable to establish IPMI v2 / RMCP+ session at /usr/lib/os-autoinst/backend/ipmi.pm line 48.
```
The die-message cannot be removed because there is a die-handler function called in its baseclass:

```
$SIG{__DIE__} = \&die_handler;
...
sub die_handler ($msg) {
    chomp($msg);
    bmwqemu::fctinfo "Backend process died, backend errors are reported below in the following lines:\n$msg";
    bmwqemu::serialize_state(component => 'backend', msg => $msg);
    $backend->stop_vm();
    $backend->close_pipes();
}
```

Reply:
We could add a list of mask-able strings into the logger and then replace them here: https://github.com/os-autoinst/os-autoinst/blob/af9bd7805f9b3b51b3516c0334f72cb162605079/log.pm#L24
But this is nothing I would do in this PR and maybe it will then again hit the timing concerns you had in the past...

A nitty hint, you could write it with a map as well..
```suggestion
    die join(' ', map { $_ eq $bmwqemu::vars{IPMI_PASSWORD} ? "[masked]" : $_ } @cmd) . ": $stderr" unless ($ret);
```